services/shuttle-crontab/src/lib.rs

```diff
@@ -0,0 +1,163 @@
+use std::str::FromStr;
+use std::sync::Arc;
+
+use axum::Router;
+use chrono::Utc;
+use cron::Schedule;
+use router::make_router;
+use serde::{Deserialize, Serialize};
+use shuttle_persist::PersistInstance;
+use shuttle_runtime::tracing::{debug, info};
+use tokio::{
+    sync::mpsc::{self, Receiver, Sender},
+    sync::oneshot,
+    time::sleep,
+};
+
+mod error;
+use error::CrontabServiceError;
+
+mod router;
+
+pub type ShuttleCrontab = Result<CrontabService, shuttle_runtime::Error>;
+
+type Responder<T> = oneshot::Sender<Result<T, CrontabServiceError>>;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct RawJob {
+    schedule: String,
+    url: String,
+}
+
+#[derive(Debug)]
+enum Msg {
+    NewJob(RawJob, Responder<()>),
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Crontab {
+    jobs: Vec<RawJob>,
+}
+
+#[derive(Debug)]
+struct CronJob {
+    schedule: Schedule,
+    url: String,
+}
+
+impl CronJob {
+    fn from_raw(raw: &RawJob) -> Self {
+        let schedule = Schedule::from_str(&raw.schedule).expect("Failed to parse schedule");
+        Self {
+            schedule,
+            url: raw.url.clone(),
+        }
+    }
+
+    async fn run(&self) {
+        debug!("Running job for: {}", self.url);
+        while let Some(next_run) = self.schedule.upcoming(Utc).next() {
+            let next_run_in = next_run
+                .signed_duration_since(chrono::offset::Utc::now())
+                .to_std()
+                .unwrap();
+            sleep(next_run_in).await;
+
+            info!("Calling {}", self.url);
+
+            let req = reqwest::get(self.url.clone()).await.unwrap();
```

Comment:
Comments about customizing this Shuttle service were given and I agree with them to some extent. The main point I would add is that this resource feels more like a `shuttle-http-request-scheduler`. A long name, I know, but if we stick with the `shuttle-crontab` name this would be misleading. The service is specialized. In addition, I would add the ability to schedule a request on the service instead of giving it just a `url` and a `schedule`.

As for a `shuttle-crontab` or `shuttle-crond`, as @jonaro00 said, I would make it more generic, as in making it able to schedule any job. I would be happy though to have a `shuttle-http-request-scheduler` in shuttle. I am curious about anyone's thoughts on this.

Reply:
Agreed. In its current form, I wouldn't say it is a relevant crate to publish, it would however be a great example in `shuttle-examples:custom-service/crontab/request-scheduler` (and why not add to the docs, hint: https://github.com/shuttle-hq/shuttle-docs/issues/86).
And by the way, now that I look, it surprises me how none of the Custom Services from the docs are not present in the examples repo (!) I was already planning on adding custom services there later, such as crontab, but others can take the opportunity if they want.

Reply:
maybe the request part should be just example how to use the cron service, cause doing a proper "cron webhook" service (with all the options like basic auth, token auth, custom headers, custom methods, retry policy with backoff) it will be a much larger job (for sure not a small tag on product board) and not sure there's a high demand for a feature like that?

Reply:
Ok, from my point of view, I would propose the following:

1. Rename this to `request-scheduler` and merge it (here or in shuttle-examples), closing the original issue.
2. Open a new issue on adding a `shuttle-crond`, the design and scope of which to be discussed separately.
3. Discuss both points above on an in-person call on Discord.

The last point may be the most important one as it is becoming a bit disheartening for @urbaniak and I to keep up with what feels like imprecise and changing requirements. I think a design session with everybody who has a vision for this to define clear requirements would help a lot.

I will ping you guys on Discord to see if we can make that happen later today!

Reply:
Thanks for taking the lead on clarifying this @morlinbrot and @urbaniak! These conversations are needed, especially because we don't have shuttle charter just yet, but this might change soon.

I concur with the points made by @jonaro00 and @urbaniak on having the `request-scheduler` as an example instead of a shuttle service for the time being. What concerns me the most is the lack of demand for such a service (hence the lack of clear requirements). 

What's reassuring though is that IMO we have a precedent that started with `shuttle-poise`, an opinionated Discord bot framework. We can chat more in our Discord in-person call today.